packages/earth-admin/src/pages-client/layers/new.tsx

```diff
@@ -0,0 +1,302 @@
+/*
+  Copyright 2018-2020 National Geographic Society
+
+  Use of this software does not constitute endorsement by National Geographic
+  Society (NGS). The NGS name and NGS logo may not be used for any purpose without
+  written permission from NGS.
+
+  Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+  this file except in compliance with the License. You may obtain a copy of the
+  License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed
+  under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+  CONDITIONS OF ANY KIND, either express or implied. See the License for the
+  specific language governing permissions and limitations under the License.
+*/
+
+import React, { useState } from 'react';
+import { navigate } from 'gatsby';
+import { JSHINT } from 'jshint';
+import { Controller, useForm } from 'react-hook-form';
+import Select from 'react-select';
+
+import { Spinner, AsyncSelect } from '@marapp/earth-components';
+import { useAuth0 } from 'auth/auth0';
+import { getAllLayers, getUniqueSlug, addLayer } from 'services/layers';
+import { noSpecialCharsRule, alphaNumericDashesRule, setupErrors } from 'utils/validations';
+
+import { LinkWithOrg } from 'components/link-with-org';
+import { ErrorMessages } from 'components/error-messages';
+import { Card } from 'components/card';
+import { Input } from 'components/input';
+import { HtmlEditor } from 'components/html-editor';
+import { JsonEditor } from 'components/json-editor';
+import { ContentLayout } from 'layouts';
+import {
+  LAYER_CATEGORY_OPTIONS,
+  LAYER_TYPE_OPTIONS,
+  LAYER_PROVIDER_OPTIONS,
+} from './model';
+import { CUSTOM_STYLES, SELECT_THEME } from '../../theme';
+
+export function NewLayer(path: any) {
+  const {selectedGroup} = useAuth0();
+
+  const {getValues, register, watch, formState, errors, setValue, control} = useForm({
+    mode: 'onChange',
+  });
+
+  const {touched, dirty, isValid} = formState;
+  const renderErrorFor = setupErrors(errors, touched);
+
+  const watchName = watch('name');
+  const watchJson = watch('config');
+  const watchCategory = watch('category');
+  const watchType = watch('type');
+  const watchProvider = watch('provider');
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [serverErrors, setServerErrors] = useState([]);
+  const [jsonError, setJsonError] = useState(false);
+  const [layerConfig, setLayerConfig] = useState({});
+
+  const [references, setReferences] = useState();
+
+  const flattenArrayForSelect = (e, field) => {
+    return !!e ? e.map(val => val[field]) : e;
+  };
+
+  const flattenObjectForSelect = (e) => {
+    return !!e ? e.value : e;
+  };
+
+  async function onSubmit(e) {
```

Comment:
use `onSubmit` on the form, not on button click
Eg: `<form onSubmit={handleSubmit(onSubmit)} />`

If you do so, you receive the form values as the first param `onSubmit(values)` which allows you to remove the getValues call

see https://github.com/natgeosociety/marapp-frontend/blob/develop/packages/earth-admin/src/pages-client/organizations/new.tsx#L44